components/global_header/hooks.tsx

```diff
@@ -23,15 +26,35 @@ export const useProducts = (): ProductComponent[] | undefined => {
 };
 
 /**
- * Hook that alerts clicks outside of the passed ref.
+ * Hook that alerts clicks outside of the passed ref or refs array.
  */
-export function useClickOutsideRef(ref: MutableRefObject<HTMLElement | null>, handler: () => void): void {
+export function useClickOutsideRef(
+    refOrRefs:
+    | MutableRefObject<HTMLElement | null>
+    | Array<MutableRefObject<HTMLElement | null>>,
+    handler: () => void,
+): void {
     useEffect(() => {
         function onMouseDown(event: MouseEvent) {
-            const target = event.target as any;
-            if (ref.current && target instanceof Node && !ref.current.contains(target)) {
-                handler();
+            const target = event.target;
+            if (!(target instanceof Node)) {
+                return;
             }
+
+            if (!Array.isArray(refOrRefs)) {
+                if (!refOrRefs.current?.contains(target)) {
+                    handler();
+                }
+                return;
+            }
+
+            const someRefContainTarget = refOrRefs.filter((ref) => ref.current).some((ref) => ref.current?.contains(target));
```

Comment:
current should always be present after you filtered out those that are not

```suggestion
            const someRefContainTarget = refOrRefs.filter((ref) => ref.current).some((ref) => ref.current.contains(target));
```